src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -145,7 +145,27 @@ public static async ValueTask<FormulaValue> DropColumns(EvalVisitor runner, Eval
             var rows = await LazyDropColumnsAsync(runner, context, sourceArg.Rows, recordIRContext, args.Skip(1).ToArray()).ConfigureAwait(false);
 
             return new InMemoryTableValue(irContext, rows);
-        }
+        }
+
+        public static async ValueTask<FormulaValue> ShowColumns(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            var tableType = ((TableValue)args[0]).Type;
+            var columnsToRemain = args.Skip(1).Select(fv => ((StringValue)fv).Value).ToArray();

```

Comment:
>ToArray() [](http://example.com/codeflow?start=85&length=9)

Instead of an array, make it a HashSet to make it more performant. That will make the next line from O(n^2) to O(n) (assuming a constant lookup in the hashset)